### ISSUE TITLE
Cargo monitor is keeping all my purchased commodities  in the list until I restart Elite.

### ISSUE THREAD

**Issue by rmb4253** [EDCD/EDDI#2612]
I buy a commodity at a station - say, gold.  Go to another station where EDDI announces that I can sell my gold for a profit of xxx credits.  I sell the gold and buy a different commodity - say Rutile expecting profit of yyy credits.  Go to another station where EDDI announces that I can sell my Gold and my Rutile for a profit of xxx + yyy.  Each time I buy a commodity, it gets added to the cargo monitor list but it never gets removed when I sell it.  I cannot edit the cargo monitor manually, it can only be done by restarting Elite.

It appears that something is stuck in my PC somewhere preventing the deletion of my sold cargo.

Windows 10 64bit version 22H2. Intel core i7-6700. 16 GB RAM.  Using EDDI plug-in and Voice Attack v 1.12 64bit.  Its been working fine for many years.

**Comment by Tkael** [comment]
That's odd. Does the problem persist after you restart EDDI?
Can you check whether the cargo.json file at `[Saved Games]\Frontier Developments\Elite Dangerous\` is being updated when you sell cargo?
Have you looked at the .log files at `%appdata%/EDDI`?

**Comment by rmb4253** [comment]
Restarting Elite seems to give me a clean sheet regarding commodities.  It only goes wrong once I started buying and selling.
Looking at the log files, there were several entries showing access denied to various json files including cargo.json.  I had to start Elite and do a couple of trades to check for cargo.json updating and can confirm that it is updating both on buying and selling.

While I was there, I noticed that EDDI was giving the correct speech information about what I had to sell, so somehow it seems to have sorted itself out.  I did another couple of trade runs and all seems to be working as expected.  A quick check of today's EDDI.log has no mention of cargo.json at all.  Perhaps me opening the file to see what was in it somehow enabled access to the file?

I'm quite happy for you to mark this issue as closed.
Thanks for the quick response.

[eddi2.log](https://github.com/EDCD/EDDI/files/15369595/eddi2.log)
